src/main/java/org/spongepowered/api/util/bans/BanBuilder.java

```diff
@@ -0,0 +1,29 @@
+package org.spongepowered.api.util.bans;
+
+import org.spongepowered.api.entity.player.User;
+import org.spongepowered.api.util.command.CommandSource;
+
+import javax.annotation.Nullable;
+import java.util.Date;
+
+/**
+ * Represents a builder that creates bans.
+ */
+public interface BanBuilder {
+
+    /**
+     * Sets the user to be banned.
+     *
+     * @param user The user
+     */
+    void setUser(User user);
+
+    void setReason(String reason);
+
+    Date setBanDate();
+
+    Date setExpirationDate();
+
+    void setSource(@Nullable CommandSource source);
+
+}
```

Comment:
Is this correct layout for a builder? I thougth it was like
.user(User).reason(Reason).banDate(Date) etc.

And why do setBanDate() and setExpirationDat() have no arguments? Change it to something like setBanDate(Date)